### PR TITLE
Ssee 893

### PR DESCRIPTION
**Pull request by frobian21**

### CODE REVIEW

File: test/uk/gov/hmrc/entrydeclarationstore/services/SubmissionReplayServiceSpec.scala [hunk 1]
```diff
@@ -143,14 +159,14 @@ class SubmissionReplayServiceSpec
...
             MockReportSender.sendReport(errorReport) returns Future.successful((): Unit)
-
             val successReport = submissionSentToEISReport(subId2, None)
             MockEntryDeclarationRepo
               .lookupMetadata(subId2)
               .returns(Future.successful(Right(replayMetadata(subId2))))
             MockEisConnector.submitMetadata(metadata(subId2), bypassCircuitBreaker = true) returns Future.successful(
               None)
             MockReportSender.sendReport(successReport) returns Future.successful((): Unit)
+            MockEntryDeclarationRepo.setSubmissionTime(subId2, Instant.now(clock)) returns Future.successful(true)
```
Comment (jeremystone) [inline test/uk/gov/hmrc/entrydeclarationstore/services/SubmissionReplayServiceSpec.scala:169]: Not sure we want to set the submission time if we've not submitted successfully.

File: it/uk/gov/hmrc/entrydeclarationstore/repositories/EntryDeclarationRepoISpec.scala [hunk 2]
```diff
@@ -136,7 +136,7 @@ class EntryDeclarationRepoISpec
       "a document with the eori & correlationId exists in the database" must {
         "return its submissionId" in {
           await(repository.lookupSubmissionId(eori, correlationId313)) shouldBe
-            Some(SubmissionIdLookupResult(receivedDateTime.toString, housekeepingAt.toString, submissionId313))
+            Some(SubmissionIdLookupResult(receivedDateTime.toString, housekeepingAt.toString, submissionId313, None))
```
Comment (jeremystone) [inline it/uk/gov/hmrc/entrydeclarationstore/repositories/EntryDeclarationRepoISpec.scala:139]: Should include a test that is gets the submission date time when it is not none.

File: test/uk/gov/hmrc/entrydeclarationstore/services/SubmissionReplayServiceSpec.scala (hunk 1 above)
Comment (frobian21) [reply test/uk/gov/hmrc/entrydeclarationstore/services/SubmissionReplayServiceSpec.scala:169]: There are two requests here. The first, subId1, returns a 400 and doesn't set the submission time. The second, subId2, returns no errors from the connector (success case) and so calls the setSubmissionTime method

Comment (jeremystone) [reply test/uk/gov/hmrc/entrydeclarationstore/services/SubmissionReplayServiceSpec.scala:169]: Ok - my misunderstanding. That's fine.

**Review by jeremystone — APPROVED** [review]
